### PR TITLE
[Private media files] Disable privacy options when classic editor is selected

### PR DESCRIPTION
**Pull request by adamziel**
#### Changes proposed in this Pull Request

This PR is related to [private atomic sites project](https://wp.me/paObgF-Ui). It changes the site privacy picker to disable private choices when the site is still using classic editor. The rationale is that TinyMCE support is a large project on it's own and would require separate planning - so it makes sense avoid delaying the initial release and move forward without it.

Ideally, we would either:

* Support classic editor, or...
* ...tell the user they need to switch to Gutenberg if they want to have a private site

Both options would make an excellent follow-up PR.

#### Testing instructions

* Test locally, calypso.live won't work
* Choose a public atomic site with classic editor selected, go to settings/general and confirm you're **unable** to see Coming Soon and Private choices
* Choose a public atomic site with gutenberg selected, go to settings/general and confirm you're **able** to see Coming Soon and Private choices
* Confirm nothing changes for simple sites and jetpack sites

### CODE REVIEW

File: client/my-sites/site-settings/form-general.jsx [hunk 1]
```diff
@@ -357,18 +357,25 @@ export class SiteSettingsFormGeneral extends Component {
...
 		const wpcomComingSoon = parseInt( fields.wpcom_coming_soon, 10 );
 
-		const isNonAtomicJetpackSite = siteIsJetpack && ! siteIsAtomic;
+		const siteIsSimple = ! siteIsJetpack;
+		const siteIsGutenbergAtomic = siteIsAtomic && selectedEditor && selectedEditor !== 'classic';
+		const siteIsClassicAtomic = siteIsAtomic && selectedEditor === 'classic';
+		const siteIsAlreadyPrivate = blogPublic === -1;
+
+		const mayGoPrivate =
+			siteIsSimple || siteIsGutenbergAtomic || ( siteIsClassicAtomic && siteIsAlreadyPrivate );
```
Comment (jblz) [inline client/my-sites/site-settings/form-general.jsx]: Won't this result in a case like this:
* a site is private, Atomic, & Gutenberg
* Classic editor is selected
* Customer sets site to private
* The option disappears without warning

Comment (adamziel) [reply client/my-sites/site-settings/form-general.jsx]: Is it possible to go back to classic editor? If that's the case, then we should add a warning there too - saying you can't do it with a private site.

Comment (adamziel) [reply client/my-sites/site-settings/form-general.jsx]: Only HEs may move the user back to classic editor, I think the best we can do is add a warning on user RC ( C02DQP0FP-slack-1584093599198300 )

**Comment by adamziel** [comment]
@jblz There are a few possible scenarios how to move forward here

### Support private media files in classic editor

This is quite hard if even possible, think editing inline HTML, pasting, using plugins etc. It would easily add 2-3 weeks before v1 unless we'd find an easy way to iframe it with correct origin.

### Make sure private files are never displayed in classic editor

To achieve this, we need to ensure the following conditions don't happen at the same time:

* Site is atomic
* Site is private
* Site uses classic editor

That's a lot of combinations and places to add guards too, it would be overly complex - especially considering p58i-8Hc-p2. How about a third solution?

### Redirect all atomic, private, classic editor users to classic editor in /wp-admin

That's not perfect, but sounds like it's aligned with direction from p58i-8Hc-p2 - and would likely be superseded by the larger project some time soon. Let's try it!

**Pull request closed without merging**